### ISSUE TITLE
Serial Device - It lose a byte of buffer

### ISSUE THREAD

**Issue by jonhare** [openimaj/openimaj#36]
OP-37: There a wrong sentence in the Seria Device (SerialReader)

**Comment by jonhare** [comment]
Original reporter: @

**Comment by jonhare** [comment]
:  1) Buffer with 3 values:

!Screen01.jpg!

2) Before "buffer = buffer.subList"

!Screen02.jpg!

3) After "buffer = buffer.subList"

!Screen03.jpg!

4) Fix

!Screen04.jpg!

5)After Fix and Before "buffer = buffer.subList"

!Screen05.jpg!

5)After Fix and After "buffer = buffer.subList"

!Screen06.jpg!

**Comment by jonhare** [comment]
davedupplaw:  Hi Javier,

Thanks for spotting this.  I have just committed a fix to this issue, as well as a junit test for testing it.

The problem actually isn't to do with the subList() call, but to do with the while() loop above. When the buffer reached capacity one byte would be dropped (it would be read with inputstream.read() and then maxSize was checked afterwards, so that byte would be dropped).

I have also made the serial reader return the leftOvers from the stream if the input stream gets to the end.

Cheers,
Dave
